### DISCUSSION TITLE
Can we set the directory for downloaded update files when configuring the tauri updater?

### DISCUSSION THREAD

**Discussion by muwoo** [tauri-apps/tauri#9747] [Q&A] [answered]
On Windows, Tauri will automatically download an update installation package to the /AppData/Local/Temp/xxx folder. Can we modify the download directory? I do not want it to be stored in this directory.

**Comment by FabianLars** [comment]
No, this is not configurable. Feel free to open a feature request (via a github issue) but tbh i doubt this will get a high priority. Why don't you like this directory? And which would you prefer instead?

**Reply by muwoo** [reply]
@FabianLars 
Thanks again for your reply!

Because I have seen Tauri windows application updates are implemented by running NSIS setup.exe through Powershell, but Powershell.exe execution will be interrupted on some windows computers with virus protection. An error will appear: `perimission denied`.

Therefore, I need to change the directory where the update package is downloaded so that it can be downloaded to a directory without virus protection, so that this problem will not occur.

Of course, users can exclude the Tauri default update download directory in windows, but for the vast majority of users, they will not know how to do this.

There is a similar problem: https://github.com/rust-lang/rustlings/issues/604

**Reply by muwoo** [reply]
Here's a similar question: https://github.com/rust-lang/cargo/issues/11544#issuecomment-1712409665

**Reply by FabianLars** [reply, marked as answer]
Hmm, interesting. Which directory will you use instead? Or how do you know which one isn't virus protected? (just curious)

Maybe we can backport https://github.com/tauri-apps/plugins-workspace/pull/1054 to v1 which removes powershell though idk if that's enough.

**Reply by muwoo** [reply]
Although I don't know which directory has virus protection, if I can set this directory or return the location of this directory when Tauri downloads updates, so that whenever I encounter the pemission denined error, I can guide the user to manually go to the corresponding directory and double-click setup.exe to install it, so there is no problem

**Reply by muwoo** [reply]
> Maybe we can backport https://github.com/tauri-apps/plugins-workspace/pull/1054 to v1 which removes powershell though idk if that's enough.

Wow, this is great! I just read about this change, it is indeed a very undone solution, looking forward to merging it into the 1.x version soon!
